### DISCUSSION TITLE
Feedback: noUnusedImports

### DISCUSSION THREAD

**Discussion by tjmgregory** [biomejs/biome#520] [General]
Searched the repo but couldn't find anywhere gathering feedback on the `noUnusedImports` nursery rule - do shout if I'm wrong.

### Issue list
- [ ] In React projects it removes the `import React from 'react'` import required at the top of all TSX/JSX files

**Comment by ematipico** [comment]
You can check the PR: https://github.com/biomejs/biome/pull/428

We intentionally kept that because some projects might use an old version of React. Biome doesn't have information about the project yet, so we called to add it as an exception.

**Reply by tjmgregory** [reply]
So, oddly I think I'm seeing it do the opposite to the decision on the PR. We've got the legacy `import React from 'react'` throughout the codebase, and running `biome format . --write && biome check --apply-unsafe ` deletes them all 🚮

Your discussion and a cursory glance at the code sounds they should be allowed to remain, correct?

**Reply by ematipico** [reply]
I may remember it incorrectly! 😅 But yeah it's a nursery rule, so we can change the logic :) 

cc @Conaclos

**Reply by Conaclos** [reply]
In the [last comments of the PR](https://github.com/biomejs/biome/pull/428#discussion_r1341192205), I decided to remove the exception because the implementation was buggy and I didn't see the value of supporting this legacy exception. The rule is not recommended, did you enable the rule on purpose?

**Reply by tjmgregory** [reply]
Gotcha 👌

> The rule is not recommended, did you enable the rule on purpose?

You mean `noUnusedImports`? Yeh we'd enabled on purpose as since switching to Biome two weeks ago we've littered imports all over the codebase, making dead code detection harder.

Appreciate the React import is legacy, and right path here [since React 17 is to stop importing it](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports) - will convince the team 👌

**Reply by EvHaus** [reply]
Apologies. I didn't see this discussion and opened an issue for this [here](https://github.com/biomejs/biome/issues/571), as I just ran into this as well.

**Reply by Conaclos** [reply]
@EvHaus This is a legacy React feature. Do you use React 16 or below?

**Reply by ematipico** [reply]
Maybe we should add a note to the diagnostics, and warn users about that with some link, that should give users proper information

**Reply by EvHaus** [reply]
I'm on React 16, but I don't think that matters. I think it depends on which tool is running React. Running React 16 inside of Vitest, for example, still requires the import. I can create a repro if it helps.

**Comment by Conaclos** [comment]
This is now addressed. We added a new option `ignoreReact`. See #2306.

Note: This option will be released in Biome 1.7
